packages/@glimmerx/helper/src/helper.ts

```diff
@@ -54,8 +69,17 @@ class BasicHelperManager implements HelperManager<BasicHelperBucket> {
 
 const basicHelperManagerFactory = (owner: Owner | undefined) => new BasicHelperManager(owner);
 
-export function helper<T, U>(helperFunction: Helper<T, U>) {
+export function helper<P extends unknown[], N, R>(
+  helperFunction: (positional: P, named: N, options: HelperOptions) => R
+): Helper<{ Args: { Named: N; Positional: P }; Return: R }> {
```

Comment:
Do we care about supporting the `helper<SomeSignature>(...)` form? (I don't have a strong opinion, but we *do* support it in Ember's helper types.)

Reply:
My leaning was to leave it out, since there's no parallel class-based `Helper` API we're aligning `helper()` with in GlimmerX.

Reply:
Makes sense to me!